actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,482 @@
+            # figure_paths[i][0] as one of those paths (all are same)
+            this_figure_path = figure_paths[i][0]
+
+            # Get correct figure path if multiple metadata, i.e.
+            # len(figure_paths[i][0]) > 1
+            if metadata != all_metadata:
+                if isinstance(this_figure_path, list) and len(this_figure_path) > 1:
+                    multiple_paths = True
+                    this_figure_path = this_figure_path[all_metadata.index(metadata)]
+                else:
+                    multiple_paths = False
+
+            if this_figure_subplots < max_cells:
+                index = self._save_single_figure(
+                    dataset,
+                    this_figure_number,
+                    this_figure_subplots,
+                    index,
+                    metadata,
+                    all_metadata,
+                    multiple_paths,
+                    this_figure_path,
+                    feature,
+                    fig_width,
+                    fig_height,
+                )
+            else:
+                # If subplot numbers are more than a max cell number (1000),
+                # then make separate figures
+                subplot_split = [
+                    max_cells for i in range(int(this_figure_subplots / max_cells))
+                ] + [
+                    (
+                        this_figure_subplots
+                        - max_cells * int(this_figure_subplots / max_cells)
+                    )
+                ]
+
+                # pop last element if 0 (figure subplots =
+                # max_cells * int(subplots/max_cells))
+                if subplot_split[-1] == 0:
+                    subplot_split.pop()
+
+                for j, sublot in enumerate(subplot_split):
+                    # If split, add a "_{number}" before ".png" in path
+                    split_fig_path = Path(
+                        str(this_figure_path)[:-4] + "_" + str(j) + ".png"
+                    )
+                    log.info(f"Splitting this figure and saving to: {split_fig_path}")
+
+                    index, dataset = self._save_single_figure(
+                        dataset,
+                        this_figure_number,
+                        sublot,
+                        index,
+                        metadata,
+                        all_metadata,
+                        multiple_paths,
+                        split_fig_path,
+                        feature,
+                        fig_width,
+                        fig_height,
+                    )
+
+            log.info(
+                f"Completed diagnostic sheet for: {metadata} {this_figure_number} "
+            )
+
+        return dataset
+
+    @staticmethod
+    def _save_single_figure(
+        dataset: pd.DataFrame,
+        figure_number: Union[int, str],
+        number_of_subplots: int,
+        index: int,
+        metadata: str,
+        all_metadata: Union[str, list],
```

Comment:
Maybe I am missing something. I see where `metadata` comes from I don't see where `all_metadata` is initialized.